sys/targets/targets.go

```diff
@@ -619,18 +626,20 @@ func init() {
 	}
 	for _, target := range List[TestOS] {
 		if List[goos] != nil {
-			if host := List[goos][goarch]; host != nil {
+			host := List[goos][goarch]
+			if host != nil && host.PtrSize > target.PtrSize {
```

Comment:
`host != nil` is not needed now

Reply:
but what if goarch doesn't exist in List[goos]?

Reply:
But this will already happen for e.g. powerpc and riscv.
If we want to keep the host for goos, then we need different logic.

Reply:
I've changed the logic, now we first split GOARCH into the 32 and 64 bit alternatives, then pick the one that fits.

Also now it effectively disables unachievable test targets by not setting BuildOS if the host is nil. This commit makes the diff messy, so please see the commits separately.